iroha-lib/model/LibsAndClassDeclarations.h

```diff
@@ -0,0 +1,4 @@
+#include "ClassDeclarations.h"
```

Comment:
IMO that kind of header with just declarations is not helping. You are including the file 3 times. The file contains 3 headers. IMO more readable would be to remove the file and copy those includes files which contains the `#include <LibsAndClassDeclarations.h>`

Reply:
Imho my option is one of the best options in terms of omitting issues regarding include loops in c++ (diamonds, fractals).